newIDE/app/src/MainFrame/EditorContainers/HomePage/index.js

```diff
@@ -203,10 +205,20 @@ export const HomePage = React.memo<Props>(
         showGetStartedSection ? 'get-started' : 'build'
       );
 
+      // If the user logs out and is on the team view section, go back to the build section.
+      React.useEffect(
+        () => {
+          if (activeTab === 'team-view' && !authenticated) {
+            setActiveTab('build');
+          }
+        },
+        [authenticated, activeTab]
+      );
+
       return (
         <I18n>
           {({ i18n }) => (
-            <>
+            <TeamProvider>
```

Comment:
This feels like a good idea, but wondering if one day we'll need something outside of the homepage and we'll end up moving it to our never ending Provider's list anyway 😁

Reply:
That is a necessary because switching tabs on the homepage unmounts and mounts component. So switching to the TeamSection, if there is no provider above, was making the front end do 4 api calls each time.

Reply:
and if we need app-wise, it will end up in this list indeed!